src/Microsoft.OData.Edm/EdmUtil.cs

```diff
@@ -492,6 +500,82 @@ internal static string FullyQualifiedName(IEdmVocabularyAnnotatable element)
             return null;
         }
 
+        internal static string FullyQualifiedName(IEdmAnnotationsTarget annotationsTarget)
+        {
+            string qualifiedName = string.Empty;
+            List<IEdmElement> segments = annotationsTarget.TargetSegments.ToList();
+            IEdmEntityContainer container = segments[0] as IEdmEntityContainer;
+            IEdmEntityContainerElement containerElement = segments[1] as IEdmEntityContainerElement;
+
+            if (container == null || containerElement == null)
+            {
+                return null;
+            }
+
+            qualifiedName = container.FullName() + "/" + containerElement.Name;
+
+            int index = 2;
+
+            IEdmSchemaType schemaType = segments[index] as IEdmSchemaType;
+            IEdmProperty property = segments[index] as IEdmProperty;
+
+            if (schemaType != null)
+            {
+                qualifiedName = qualifiedName + "/" + GetSchemaTypeName(schemaType, segments, index);
+            }
+
+            if (property != null)

```

Comment:
```suggestion
            else if (segments[index] is IEdmPropertyType property)
```
Since `segments[index]` cannot be both an `IEdmSchemaType` and an `IEdmProperty`, I think you should use `else if` so the second condition is not evaluated if the first condition evaluated to true. In addition, depending on which scenario is more possible, determine which condition should be evaluated first.